notes.go

```diff
@@ -0,0 +1,207 @@
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.CreateNoteRequest{}
+	if err := c.ShouldBindJSON(body); err != nil {
+		writeError(c, http.StatusBadRequest, err)
+		return
+	}
+
+	res, err := h.notesClient.CreateNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) GetNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.GetNoteRequest{
+		Id: c.Param("note_id"),
+	}
+
+	res, err := h.notesClient.GetNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) UpdateNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.UpdateNoteRequest{}
+	if err := c.ShouldBindJSON(body); err != nil {
+		c.JSON(http.StatusOK, httpError{Error: err.Error()})
+		return
+	}
+	body.Id = c.Param("note_id")
+
+	res, err := h.notesClient.UpdateNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) DeleteNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.DeleteNoteRequest{
+		Id: c.Param("note_id"),
+	}
+
+	res, err := h.notesClient.DeleteNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) ListNotes(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.ListNotesRequest{
+		AuthorId: c.Param("author_id"),
+	}
+
+	res, err := h.notesClient.ListNotes(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func (h *notesHandler) ExportNote(c *gin.Context) {
+	bearer, err := authenticate(c)
+	if err != nil {
+		writeError(c, http.StatusUnauthorized, err)
+		return
+	}
+
+	body := &notesv1.ExportNoteRequest{}
+	if err := c.ShouldBindJSON(body); err != nil {
+		writeError(c, http.StatusBadRequest, err)
+		return
+	}
+	body.NoteId = c.Param("note_id")
+
+	res, err := h.notesClient.ExportNote(contextWithGrpcBearer(context.Background(), bearer), body)
+	if err != nil {
+		writeError(c, http.StatusInternalServerError, err)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
```

Comment:
Right now, you're returning a JSON object containing the raw bytes of a file. What we want, is that this endpoint returns a downloadable file to the client.

There are special HTTP headers that are required for the browser to make the client download a file on response.

You should look into putting the correct HTTP headers and transforming the gRPC request body into an actual file instead of just sending the raw bytes as JSON.

I found this on how to do this with GIN:
https://stackoverflow.com/questions/31638447/how-to-server-a-file-from-a-handler-in-golang

CC: @edouard-sn

Reply:
Totally but idk I think `c.File` is only for accessible files I guess for us it'll more be like this (`[]byte`, c.Data)
https://stackoverflow.com/questions/62003501/gin-serve-byte-as-file

Reply:
You're right! Thanks :)